spec/handlers/sessions_create_spec.rb

```diff
@@ -356,6 +356,70 @@
             end
           end
         end
+
+        context "linked to no one" do
+          context "but user already has already authenticated with the same provider before" do
+            context "new_social user" do
+              let!(:user) { FactoryGirl.create(:new_social_user) }
+              let!(:authentication) { FactoryGirl.create(:authentication, user: user) }
+
+              it "returns a status output instead of raising an exception" do
+                result = lambda { |uid|
```

Comment:
actually seeing that you use this a couple of times, just put it in a method with a descriptive name... 

```
def handle_for_uid(uid)
  described_class.handle(
    user_state: user_state,
    request: MockOmniauthRequest.new(authentication.provider, uid, {})
  )
end
```

and then

```
handle_for_uid(111)
blah = handle_for_uid(222)
```

This test code shows you can define methods inside or outside of contexts and they will have access to `let` variables.

```
require 'rspec'

describe 'something' do

  context 'blah' do
    let(:foo) { "bar" }

    it 'works' do
      expect(foo).to eq "bar"
      expect(method_inside_context).to eq "bar"
      expect(method_outside_context).to eq "bar"
    end

    def method_inside_context
      foo
    end
  end

  def method_outside_context
    foo
  end

end
```